src/main/java/org/openrewrite/hibernate/EmptyInterceptorToInterface.java

```diff
@@ -0,0 +1,91 @@
+               "This recipe migrates 5.x `Interceptor#onPrepareStatement(String)` to 6.0 `StatementInspector#inspect()`.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(new FindImplementations(EMPTY_INTERCEPTOR),
+                new JavaIsoVisitor<ExecutionContext>() {
+                    @Override
+                    public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, ExecutionContext ctx) {
+                        J.ClassDeclaration cd = super.visitClassDeclaration(classDecl, ctx);
+                        if (cd.getExtends() != null && TypeUtils.isOfClassType(cd.getExtends().getType(), EMPTY_INTERCEPTOR)) {
+                            cd = cd.withExtends(null).withImplements(ListUtils.concat(cd.getImplements(), (TypeTree) TypeTree.build("Interceptor").withType(JavaType.buildType(INTERCEPTOR)).withPrefix(Space.SINGLE_SPACE)));
+                            Boolean prepareStatement = getCursor().pollMessage("prepareStatementFound");
+                            if (Boolean.TRUE.equals(prepareStatement)) {
+                                cd = cd.withImplements(ListUtils.concat(cd.getImplements(), (TypeTree) TypeTree.build("StatementInspector").withType(JavaType.buildType(STATEMENT_INSPECTOR)).withPrefix(Space.SINGLE_SPACE)));
+                            }
+                            maybeAddImport(INTERCEPTOR);
+                            maybeAddImport(STATEMENT_INSPECTOR);
+                            maybeRemoveImport(EMPTY_INTERCEPTOR);
+                        }
+                        return autoFormat(cd, ctx);
+                    }
+
+                    @Override
+                    public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext ctx) {
+                        J.MethodDeclaration md = super.visitMethodDeclaration(method, ctx);
+                        if (md.getLeadingAnnotations().stream().anyMatch(OVERRIDE_ANNOTATION_MATCHER::matches) && md.getSimpleName().equals("onPrepareStatement")) {
+                            getCursor().putMessageOnFirstEnclosing(J.ClassDeclaration.class, "prepareStatementFound", true);
+                            String template = "@Override\n" +
+                                              "public String inspect() {\n" +
+                                              "}\n";
+                            J.MethodDeclaration inspect = JavaTemplate.builder(template)
+                                    .javaParser(JavaParser.fromJavaVersion())
+                                    .build()
+                                    .apply(getCursor(), md.getCoordinates().replace());
+                            md = inspect.withBody(md.getBody()).withModifiers(md.getModifiers()).withLeadingAnnotations(md.getLeadingAnnotations()).withParameters(md.getParameters());
```

Comment:
Do we need the `withParameters` here if that's expected to be empty?
Same for `withModifiers`; what cases do we want to cover beyond the required `public`?

Reply:
Cross-eyed review, of course yes thanks!